mlflow/paddle/_paddle_autolog.py

```diff
@@ -0,0 +1,151 @@
+from packaging.version import Version
+import logging
+import mlflow.paddle
+import os
+import shutil
+import tempfile
+import paddle
+
+from mlflow.utils.autologging_utils import (
+    ExceptionSafeAbstractClass,
+    BatchMetricsLogger,
+    MlflowAutologgingQueueingClient,
+    get_autologging_config,
+)
+
+logging.basicConfig(level=logging.ERROR)
+
+
+def _get_optimizer_name(optimizer):
+    return optimizer.__class__.__name__
+
+
+class __MLflowPDCallback(paddle.callbacks.Callback, metaclass=ExceptionSafeAbstractClass):
+    """
+    Callback for auto-logging metrics and parameters.
+    """
+
+    def __init__(self, client, metrics_logger, run_id, log_models, log_every_n_epoch):
+        self.early_stopping = False
+        self.client = client
+        self.metrics_logger = metrics_logger
+        self.run_id = run_id
+        self.log_models = log_models
+        self.log_every_n_epoch = log_every_n_epoch
+
+    def _log_metrics(self, logs, current_epoch):
+        metrics = dict(
+            map(
+                lambda x: (x[0], float(x[1][0])) if isinstance(x[1], list) else (x[0], float(x[1])),
+                logs.items(),
+            )
+        )
```

Comment:
```suggestion
    metrics = {
        key: (metrics[0] if isinstance(metrics, list) else metrics) for key, metrics in logs.items()
    }
```

Can we use dict comprehension here, which I think is simpler?

Reply:
btw when does `metrics` become a list?

Reply:
Sometimes the logs should be in the form like `{'loss': [0.6911203], 'acc': 0.6924833333333333, 'step': 1874, 'batch_size': 32}`. But the list always contains one element. Therefore, recording the element in that list should be better.